### ISSUE TITLE
Ventilation in a tunnel

### ISSUE THREAD

**Issue by LS74** [firemodels/fds#3416]
Hello,

I have a problem with my model. I simulate a tunnel (the geometry is a vault) with extraction on the ceiling. I want to control the outflow along the tunnel. So I put devc to measure the outflow differences and the outflow along the tunnel.  And I observed that the mass conservation is not respected before the beggining of the extraction, and after, the outflow differences don't correpond with the outflow extracted.

So, I tried on the simpler modeles, as : 
- Without wall, a boundary condition from one side and a vent on the other. All the conditions are respected. 
- With 4 simple walls. All the conditions are respected.
- Without wall, but with an OBST on the ceiling to perturbate the flow. All the conditions are respected. 

Can you tell me why I have these problems?

Thank you so much for your response.

Sincerly.

Lucile

[test1.txt](https://github.com/firemodels/fds-smv/files/112375/test1.txt)

**Comment by mcgratta** [comment]
There is not output. What am I supposed to look at?

**Comment by LS74** [comment]
I join the file .txt . Can you tell me if you see an error of writing or if you know why I have these problems. Knowing, more I grow the tunnel, more the error grows too. 

Ps : All the OBST seem to be right, there are lot of lines because the volume is composed of some vaults with different areas.

**Comment by mcgratta** [comment]
There are no DEVC lines. What do you want me to look at?

**Comment by LS74** [comment]
oh sorry I remove them to simplify the modele.

this one has the outputs:

[test1.txt](https://github.com/firemodels/fds-smv/files/112433/test1.txt)

**Comment by mcgratta** [comment]
Can you tell me what is wrong with these volume flows? What should I expect to see?

**Comment by LS74** [comment]
I prescribe an initial condition : vel =-5.4 m/s. From the begining of the simulation to the start of the ventilation, I expect to obtain a constant flow along the tunnel, or this is not the case. This is the first point. I would like to know why the flow is not constant. The second point is : the flow difference between each extraction is not equal to the flow extracted. Why? 
In fact my real tunnel is 1.3 km long, and with this lenght, no flow differences are correct. I suppose there is a little error which propagate and on one hundred meter it's not apparent, but on 2 kilometers the error is glaring. So I think to change maybe the initial condition with Neumann constraint? Or maybe it's cause of the lenght of the mesh, but we have to calculate only the flows.

**Comment by mcgratta** [comment]
I do not understand you. Just say something like this:

I expect Vx02 to be equal to Vx04.

**Comment by LS74** [comment]
Ah okay, I expect that all the vx are equal before the begining of the extraction.
And after, the differences between all the vx must be equal to the extract flow for each time step.

Vz is to check the outflow extraction.

**Comment by mcgratta** [comment]
Sorry. I cannot understand this at all. It is too complicated and I do not have time to work on it. Reduce the file to 20 lines or less. If you cannot show the problem in 20 lines or less, then you need to add back the OBST lines SLOWLY until the problem is shown. At that point, you will probably solve the problem yourself.

**Comment by drjfloyd** [comment]
Just taking a quick look at your smokeview file.  Your open boundary condition, tunnel inlet condition, and all your volume flow devices across the tunnel width are rectangles.  Your tunnel is not a rectangle. So a portion of all of those lie outside the tunnel.  You are not just measuring what is going on inside the tunnel, but whatever strange flowfield results outside the tunnel as well.  Use multiple VENT inputs at the tunnel ends so you only specify the inlet condition over the tunnel mouth and not beyond it. At each volume flow location use multiple DEVC lines to just get the volume flow inside the tunnel (you will have to sum the DEVC in post processing or with a CTRL.

**Comment by LS74** [comment]
Hello,

I noted your advice. And I applied it. So I come back to you because my problem is not solved. In fact, in the attachement, you will find an other simple example. In the file .xls  (DEVC.csv) : the air flux difference between each extraction (noted Vx) is different from 2*0.6m3/s (Vz, the mass air flux extraction). Why? I find again this problem for each geometry (vault, simple wall).

Thank you.

Lucile

[test.txt](https://github.com/firemodels/fds-smv/files/114057/test.txt)
[test.xlsx](https://github.com/firemodels/fds-smv/files/114053/test.xlsx)

**Comment by mcgratta** [comment]
This is not simple. I would like to see one supply and one extraction vent. Do not use U0. Use only 2 DEVC lines. The question should be -- why do the supply and extraction volume flow not match?

**Comment by LS74** [comment]
Yes this is the good and real question.

With this scenario : 
- The initial volume flow is not constant along the tunnel before the extraction,
- After the extraction, the volume flow is not constant for each time step and the difference is always not equal to the extraction.

[test.xlsx](https://github.com/firemodels/fds-smv/files/114099/test.xlsx)
[test.txt](https://github.com/firemodels/fds-smv/files/114100/test.txt)

**Comment by mcgratta** [comment]
There is a small error in the normal component of velocity at solid surfaces. To reduce this error, you can set a parameter called `VELOCITY_TOLERANCE` on the `PRES` line. Try running this case:

```
&HEAD CHID = 'test', TITLE = 'test'/
&MESH IJK= 50 35 10 XB= 0 50 -6 6 0 5 /
&TIME T_END =  600  /
&PRES VELOCITY_TOLERANCE=0.05 /
&SURF ID ='extraction_C1', VEL= 0.6, COLOR = 'LIGHT BLUE'/
&VENT XB= 25 26 3.6 3.6 2.5 3.5 COLOR='RED' SURF_ID='extraction_C1' /
&VENT XB= 25 26 -3.6 -3.6 2.5 3.5 COLOR='RED' SURF_ID='extraction_C1' /
&OBST XB= 0.000, 512.000, -7.000,  6.000, 4.000, 4.300 , COLOR = 'GRAY'/
&OBST XB= 0.000, 512.000, -7.000, -3.600, 0.250, 4.000 , COLOR = 'GRAY'/
&OBST XB= 0.000, 512.000,  3.600,  6.000, 0.250, 4.000 , COLOR = 'GRAY'/
&OBST XB= 0.000, 512.000, -7.000,  6.000, 0.000, 0.250 , COLOR = 'GRAY'/
&VENT XB= 50 50 -7 6 0 4 ,SURF_ID='OPEN' /
&SURF ID = 'CL', VEL =-4.9, TMP_FRONT = 0.0, COLOR = 'LIGHT BLUE'/
&VENT XB= 0 0 -4.0 4.0 0.25 4.00, SURF_ID = 'CL'/
&DEVC XB= 25 26  3.60  3.60 2.50 3.50 QUANTITY='VOLUME FLOW' ID='Vz233 '/
&DEVC XB= 25 26 -3.60 -3.60 2.50 3.50 QUANTITY='VOLUME FLOW' ID='Vz233 '/
&DEVC XB= 50 50 -3.60  3.60 0.25 4.00 QUANTITY='VOLUME FLOW' ID='Vx364 '/
&DEVC XB= 0 0   -3.60  3.60 0.25 4.00 QUANTITY='VOLUME FLOW' ID='Vx368 '/
&TAIL /
```

Search the FDS User's Guide for `VELOCITY_TOLERANCE` to learn more. Another way to reduce this kind of error is to align the walls of the tunnel with the exterior of the computational domain.

**Comment by LS74** [comment]
I applied VELOCITY_TOLERANCE and I changed the initial condition : I remplaced VELOCITY by a PRESSURE condition, the computations are running. But they seem to give great results.

Thank you so much for your disponibility and your help.

However, I would like have one last question. I want to run the computations on Linux with fds6. I saw that the command lines which run the computations have changed. I tried : 

fds job_name.fds >& job_name.err &

but there's nothing happening. What are the good commands which permit to run the compute? 

Best regards.
